bridges/snowbridge/primitives/router/src/outbound/mod.rs

```diff
@@ -175,8 +180,12 @@ impl<'a, Call> XcmConverter<'a, Call> {
 	}
 
 	fn convert(&mut self) -> Result<(AgentExecuteCommand, [u8; 32]), XcmConverterError> {
-		// Get withdraw/deposit and make native tokens create message.
-		let result = self.native_tokens_unlock_message()?;
+		let result = match self.peek() {
+			Ok(Transact { .. }) => self.transact_message()?,
+			// Get withdraw/deposit and make native tokens create message.
+			Ok(WithdrawAsset { .. }) => self.native_tokens_unlock_message()?,
+			_ => return Err(XcmConverterError::UnexpectedInstruction),
```

Comment:
Forward the underlying error if there is one.
```suggestion
			Err(e) => e,
			_ => return Err(XcmConverterError::UnexpectedInstruction),
```

Reply:
https://github.com/Snowfork/polkadot-sdk/pull/116/commits/e8c9324ec4b9f93ce9594644a35257d979f83b3a